### ISSUE TITLE
Unable to Select Recipients in Mailing with Version 2.9

### ISSUE THREAD

**Issue by bmango** [veda-consulting/uk.co.vedaconsulting.mosaico#499]
Since upgrading to version 2.9 it is no longer possible to select recipients on either the Mosaico mailing screen or the traditional mailing screen. I have replicated this on the demo site. The error that comes up in the console (of the demo site) is:

```
angular.js:15697 ReferenceError: date is not defined
    at HTMLInputElement.<anonymous> (angular-modules.bf52…1600c52d.js:1984:20)
    at HTMLDivElement.dispatch (jquery.min.js:3:12369)
    at HTMLDivElement.r.handle (jquery.min.js:3:9088)
    at Object.trigger (jquery.min.js:3:11488)
    at HTMLInputElement.<anonymous> (jquery.min.js:3:18865)
    at Function.each (jquery.min.js:2:2903)
    at n.fn.init.each (jquery.min.js:2:846)
    at n.fn.init.trigger (jquery.min.js:3:18841)
    at n.fn.init.n.fn.<computed> [as change] (jquery.min.js:4:16184)
    at Object.ngModel.$render (angular-modules.bf52…1600c52d.js:5452:45)
angular.js:15697 

Possibly unhandled rejection: {"error_code":0,"entity":"Mailing","action":"create","is_error":1,"error_message":"Mailing has not been saved, Content maybe out of date, please refresh the page and try again"}
angular-modules.bf52…c52d.js?r882b5:2260 

Uncaught ReferenceError: results is not defined
    at Object.results (angular-modules.bf52…1600c52d.js:2260:23)
    at Object.success (select2.min.js:30:314)
    at Object.<anonymous> (jquery.min.js:2:28676)
    at i (jquery.min.js:2:27362)
    at Object.fireWith [as resolveWith] (jquery.min.js:2:28121)
    at x (jquery.min.js:4:22575)
    at XMLHttpRequest.c (jquery.min.js:4:26776)
```

**Comment by KilakwaBT** [comment]
This is what we see when trying to select a group. This was an existing mailing which already had a group selected and we tried to add an additional group. The behavior is the same with new mailings.
![image](https://user-images.githubusercontent.com/28601657/156782089-1d5bc166-19d5-4fd5-b108-4429610de5e8.png)

**Comment by MegaphoneJon** [comment]
This is fixed in Civi 5.47 - and in my opinion Mosaico 2.9 should be pulled for versions below 5.47.  See here: https://github.com/veda-consulting-company/uk.co.vedaconsulting.mosaico/issues/494

You can apply the patch to core to avoid rolling back, though you can roll back to Mosaico 2.8 safely.

**Comment by bmango** [comment]
Thanks for your answer @MegaphoneJon and apologies for duplicating this issue (#494)). I always forget that GitHub by default shows only open issues so hadn't checked the closed issues!
